### DISCUSSION TITLE
Add the possibility to update client proxy

### DISCUSSION THREAD

**Discussion by andrelambru** [encode/httpx#3311] [General] [answered]
There isn't a way to edit a Client() proxy without recreating it (and loosing the session).

On request module we have session.update(proxies=proxies) for example.

**Comment by lovelydinosaur** [comment, marked as answer]
Duplicate of https://github.com/encode/httpx/discussions/2618

**Reply by karpetrosyan** [reply]
Also, a good reference is: https://github.com/encode/httpx/discussions/3293#discussioncomment-10545110

**Reply by lovelydinosaur** [reply]
Thank you @karpetrosyan. ☺️

**Reply by karpetrosyan** [reply]
It seems like the question about dynamically changing transports/proxies is one of the common ones. Maybe we should mention it in the documentation as an example of transport.

**Reply by lovelydinosaur** [reply]
Neat thought, yep... if we did want to do this, let's start by opening a documentation issue for it.
Could be problematic that proxy rotation might typically be used to spam requests, I'd rather we didn't advocate for that.

**Reply by andrelambru** [reply]
> Neat thought, yep... if we did want to do this, let's start by opening a documentation issue for it. Could be problematic that proxy rotation might typically be used to spam requests, I'd rather we didn't advocate for that.

That's true but if someone wants to spam he can already do with rotating proxies.

My use-case is web automation.
